### PR TITLE
feat(STN-136):  polish horizontal card

### PR DESCRIPTION
**Pull request by silverli**
# READY FOR REVIEW 🛑 

## Summary
**Card**: [STN-136](https://sparkbox.atlassian.net/browse/STN-136)
**Author**: @meranicosme 

Polishes and adds refinements for horizontal card, external link, and image zoom hover effect.

## Steps to Test
1. Check out the horizontal card locally. Make sure behavior and style is as expected.
2. In the `humsci_basic` directory, run `npm test` and make sure everything passes.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss [hunk 1]
```diff
@@ -16,6 +16,7 @@
     font-weight: hb-theme-font-weight(semibold);
     font-size: 2.0625rem;
     line-height: 125%;
+    color: red;
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss]: This is great!!! :smile: `color` and `background-color` are so good for testing!

File: docroot/themes/humsci/humsci_basic/src/scss/elements/_base.scss [hunk 2]
```diff
@@ -64,17 +64,45 @@ p:not([class]) {
...
+    background-size: 100% 200%;
+    background-position: 0 -100%;
+    background-repeat: no-repeat;
+    transition: background-position ease 300ms;
+
+    &:hover {
+      background-position: 0 -60%;
+
+      .fa-ext::before {
+        background: svg('<svg viewBox="0 0 12 12"><path d="M11.273 5.818v5.455H.727V.727h5.091m1.455 0h4v4m0-4L6.182 5.818" fill="transparent" stroke="#{$hb-color--primary}" stroke-width="2" stroke-miterlimit="10"/></svg>') center center no-repeat;
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/elements/_base.scss]: Targeting the stroke color in the SVG. Bravo! :tada:

File: docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss [hunk 3]
```diff
@@ -105,7 +105,7 @@
         border-bottom: 0 none;
 
         img {
-          transform: scale(1.2);
+          transform: scale(1.07);
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss:108]: Nice!! This looks great! :smile:

File: docroot/themes/humsci/humsci_basic/src/scss/components/_node--news-item 2.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  }
+
+  .field-hs-news-date {
+    line-height: 1.4;
+  }
+
+  .field-hs-news-link {
+    line-height: 1.4;
+  }
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_node--news-item 2.scss]: Just an FYI. You don't need to make any changes as there are multiple ways to do the same thing. :wink:
Lines 50 through 56 can also be written like this:
```
.field-hs-news-date,
.field-hs-news-link {
    line-height: 1.4;
}
```

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss [hunk 5]
```diff
@@ -119,3 +126,25 @@
...
+//
+//     &:hover,
+//     &:focus {
+//       color: hb-colorful-variation(primary);
+//       border-bottom-color: hb-colorful-variation(primary);
+//       background-position: 0 -60%;
+//       // box-shadow: inset 0 -0.5625rem 0 lighten(hb-colorful-variation(primary), 70%);
+//     }
+//   }
+// }
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss]: Clean up task --> remove commented code on lines 130 to 150. (I do this too! 🙈)

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.general.scss [hunk 6]
```diff
@@ -11,3 +11,13 @@
...
+
+// Other one-off applications of @include hb-page-width.
+// If a component needs *any* additional styles, move the rule into /src/scss/components.
+.layout-builder__message {
+  @include hb-page-width;
+}
+
+.layout-builder-form .form-actions {
+  @include hb-page-width;
+}
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.general.scss]: Lines 15 through 23 look to be duplicated styles. We can remove them.

Here's the output from `npm run test`
npm run test

File: docroot/themes/humsci/humsci_basic/src/scss/elements/_base.scss [hunk 7]
```diff
@@ -61,21 +61,42 @@ p:not([class]) {
   margin-top: 0;
 }
 
+.field-hs-news-link a {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/elements/_base.scss]: Let's move the styles for `.field-hs-news-link a` into `docroot/themes/humsci/humsci_basic/src/scss/components/_node--news-item.scss` alongside the other `.field-hs-news` classes.

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.svg 2.scss [hunk 8]
```diff
@@ -0,0 +1,51 @@
+//  Helper function to replace characters in a string
+@function str-replace($string, $search, $replace: '') {
+  $index: str-index($string, $search);
+  @return if($index, str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace), $string);
+}
+
+
+// SVG URL fixer Icons by Taylor Hunt
+// http://codepen.io/tigt/post/optimizing-svgs-in-data-uris
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.svg 2.scss]: Love this! Taylor Hunt is a CSS guru.

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss [hunk 9]
```diff
@@ -89,21 +89,28 @@
...
+
+  &:hover,
+  &:focus {
+    box-shadow: inset 0 -0.5625em 0 $box-shadow-color;
+    transition-delay: 0s, 0s;
+    border-bottom-color: $border-color;
+  }
+}
+
 @mixin hb-link {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.fonts.scss:106]: I'm seeing a different link style on headings. This is what I see.
heading

I _think_ we need to include this mixin on `hb-card__title` headings with links. That would affect all cards that have headings with links. Let's look at this together just to make sure. 👀

**Review by heatherdesigns — CHANGES_REQUESTED** [review]
very nice

Great work tackling these refinements! :tada:

I left some comments throughout the code base. I'm available to answer any questions you have and/or to pair.

I also noticed some duplicates of existing files. Perhaps something happened during a rebase. 🤷‍♀It happens. :smile: I'm including screenshots below of the duplicate files. Look for the files with `2` added to the file name at the end. Make sure all of your changes are in the original files, then delete the duplicate files.

grid duplicates

component duplicates

tools duplicate

File: docroot/themes/humsci/humsci_basic/templates/layouts/three_column_w_image.html.twig [hunk 10]
```diff
@@ -8,6 +8,14 @@
...
 {% endif %}
 
+{% if content.left_sidebar and content.right_sidebar %}
+  {% set attributes = attributes.addClass(['hb-obj-three-column-w-image--two-sidebar']) %}
+{% elseif content.left_sidebar or content.right_sidebar %}
+  {% set attributes = attributes.addClass(['hb-obj-three-column-w-image--one-sidebar']) %}
+{% else %}
+  {% set attributes = attributes.addClass(['hb-obj-three-column-w-image--no-sidebar']) %}
+{% endif %}
+
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/templates/layouts/three_column_w_image.html.twig]: Sorry I didn't catch this earlier! It looks like lines 11 through 18 are duplicated.

**Comment by meranicosme** [comment]
Created new PR due to branch rebasing conflicts, with past sprint.

**Pull request closed without merging**